src/main/python/resources/annotaters/annotator.py

```diff
@@ -0,0 +1,39 @@
+from src.main.python.core.logger import Logger
+from src.main.python.core.global_manager import CapiceManager
+from src.main.python.resources.annotaters.manual import ManualAnnotator
+from src.main.python.resources.annotaters.lookup import FastaLookupAnnotator
+import pandas as pd
+
+
+class Annotator:
+    def __init__(self, dataset: pd.DataFrame):
+        self.log = Logger().logger
+        self.manager = CapiceManager()
+        self.fasta_lookup = FastaLookupAnnotator()
+        self.manual_annotater = ManualAnnotator()
+        self.dataset = dataset
+
+    def annotate(self):
+        """
+        Start the annotation process.
+        :return: pandas dataframe
+        """
+        self.log.info('Starting manual annotation process.')
+        self._add_sequence()
+        self.dataset = self.manual_annotater.process(dataset=self.dataset)
+        self.log.info('Annotation successful.')
+        self.log.debug(
+            'Final shape of the annotated data: {}'.format(self.dataset.shape)
+        )
+        return self.dataset
+
+    def _add_sequence(self):
+        self.log.debug('Annotation addition: sequence')
+        self.dataset['Seq'] = self.dataset.apply(
+            lambda x: self.fasta_lookup.get_reference_sequence(
+                chromosome=x['Chr'],
+                start=x['Pos'] - 75,
+                end=x['Pos'] + 75
+            ), axis=1
+        )
+        self.fasta_lookup.close_connection()
```

Comment:
So I would just need a `def __exit__(self, type, value, traceback):` in this case, since I don't handle opening the file itself (that is something that pysam does)

Reply:
The main goal is to ensure `fasta.close()` (from `FastaLookupAnnotator().close_connection()`) is also called in case an exception occurs. Mainly used try-finally myself, though `__enter__` & `__exit__` seemed like a better solution for the current design. I imagine something like `FastaLookupAnnotator()._load_fasta()` becoming `__enter__()`, `FastaLookupAnnotator().close_connection()` becoming `__exit__()` and then calling it within `Annotator.annotate()` through something like:
```
with self.fasta_lookup as lookup:
  lookup.get_reference_sequence(
                chromosome=x['Chr'],
                start=x['Pos'] - 75,
                end=x['Pos'] + 75
            )
```
At least, that's my theory.

This would ofcourse result in the loading of the fasta file being done during `Annotator.annotate()` instead of `Annotator.__init__()` so errors caused by loading the fasta would occur later. On the other hand, best practice indicates not keeping file connections open longer than actually needed (except if multiple calls would be made to the same file).

Reply:
I will take note of this and keep it in mind in cade `LookupAnnotator.py` becomes available again.